iocage/lib/ioc_create.py

```diff
@@ -86,6 +86,106 @@ def create_jail(self):
             iocage.lib.ioc_destroy.IOCDestroy().destroy_jail(location)
             sys.exit(1)
 
+    def _get_template_config_fn(self, jail_uuid, path):
+        config_fn = iocage.lib.ioc_json.IOCJson(path).json_get_value
+        try:
+            config_fn("cloned_release")
+        except (IOError, OSError, FileNotFoundError, UnboundLocalError):
+            # Unintuitevly a missing template will throw a
+            # UnboundLocalError as the missing file will kick the
+            # migration routine for zfs props. We don't need that :)
+            raise RuntimeError(f"Template: {self.release} not found!")
+
+        return config_fn
+
+    def _get_clone_config_fn(self, jail_uuid, path):
+        config_fn = iocage.lib.ioc_json.IOCJson(path).json_get_value
+        try:
+            config_fn("cloned_release")
+        except (IOError, OSError, FileNotFoundError, UnboundLocalError):
+            if os.path.isdir(f"{self.iocroot}/templates/{self.release}"):
+                iocage.lib.ioc_common.logit({
+                    "level"  : "EXCEPTION",
+                    "message": "You cannot clone a template, "
+                               "use create -t instead."
+                },
+                    _callback=self.callback,
+                    silent=self.silent)
+            else:
+                # Yep, self.release is actually the source jail.
+                iocage.lib.ioc_common.logit({
+                    "level"  : "EXCEPTION",
+                    "message": f"Jail: {self.release} not found!"
+                },
+                    _callback=self.callback,
+                    silent=self.silent)
+        return config_fn
+
+    def _get_release_config(self, jail_uuid, path):
```

Comment:
this one's tricky, as it's very different from all the other ones.

i'm not calling it a smell, just, like, production code that deals with realities ;)